api/server.go

```diff
@@ -38,12 +48,66 @@ func (s *Server) walletBalanceHandler(w http.ResponseWriter, req *http.Request,
 	})
 }
 
+func (s *Server) walletAddressHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	writeJSON(w, Address{s.w.NextAddress()})
+}
+
+func (s *Server) walletAddressesHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	writeJSON(w, Addresses{s.w.Addresses()})
+}
+
+func (s *Server) walletTransactionsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	writeJSON(w, Transactions{s.w.Transactions()})
+}
+
+func (s *Server) walletSendSiacoinsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var err error
+	defer writeJSON(w, SendReceipt{
+		Error: &err,
+	})
+
+	var out types.SiacoinOutput
+	if err := json.NewDecoder(req.Body).Decode(&out); err != nil {
+		return
+	}
+	txn := types.Transaction{
+		SiacoinOutputs: []types.SiacoinOutput{out},
+	}
+	if toSign, _, err := s.w.FundTransaction(&txn, out.Value, nil); err != nil {
+		return
+	} else if err := s.w.SignTransaction(*s.vc, &txn, toSign); err != nil {
+		return
+	}
+	s.s.BroadcastTransaction(txn, nil)
+}
+
+func (s *Server) walletBroadcastTransactionHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var err error
+	defer writeJSON(w, SendReceipt{
+		Error: &err,
+	})
+
+	var txn types.Transaction
+	if err := json.NewDecoder(req.Body).Decode(&txn); err != nil {
+		return
+	}
+	// TODO: Validate?
+	s.s.BroadcastTransaction(txn, nil)
```

Comment:
ah -- in addition to broadcasting the transaction, we also want to add it to our own txpool. That takes care of validation too, since `tp.AddTransaction` will validate the transaction before adding it.